Fix BigInt.asUintN and BigInt.asIntN implementation (fixes #1573)

anivar opened the pull request:
## Summary

Fixes issue #1573 where BigInt.asUintN(64, -1n) incorrectly returned -1n instead of the expected 18446744073709551615n.

## Problem

The previous implementation used byte array manipulation which had several issues:
- For larger bit widths, it would return the original BigInt without truncation
- The modular arithmetic was incorrect for negative numbers
- The algorithm did not follow the ECMAScript specification

## Solution

Replaced the byte array approach with proper modular arithmetic following the ECMAScript specification:

- BigInt.asUintN(bits, bigint): Returns bigint modulo 2^bits as unsigned integer
- BigInt.asIntN(bits, bigint): Returns signed representation using two's complement

## Changes

- Implemented correct ECMAScript algorithm using BigInteger.remainder() and bit shifting
- Refactored code into asUnsignedN() and asSignedN() helper methods for clarity  
- Added comprehensive test cases covering edge cases and boundary conditions
- Removed unused import (java.util.Arrays)

## Test Results

Before the fix:
BigInt.asUintN(64, -1n) === -1n  // Wrong

After the fix:
BigInt.asUintN(64, -1n) === BigInt('0xffffffffffffffff')  // Correct

All existing tests pass, and new comprehensive tests validate the fix.

Fixes #1573

rbri commented:
@anivar thanks a lot for this.

Can you please make sure this PR only contains one chance - looks like you build this on top of the ISODate stuff.

anivar commented:
@rbri Thanks for the review\! I've cleaned up the PR to only include the BigInt.asUintN and BigInt.asIntN fixes.

The PR now contains only:
- Fixed implementation in NativeBigInt.java using proper modular arithmetic
- Comprehensive tests in BigIntTest.java covering the reported issue and edge cases

The ISO date parsing changes have been removed and will be submitted as a separate PR.

Ready for review\!

rbri commented:
LGTM +1 for merging

gbrail commented:
Thanks -- looks good!
